modules/api/php/endpoints/candidate/visit/visit.class.inc

```diff
@@ -171,10 +174,120 @@ class Visit extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PATCH request.
+     * For now, only support a NextStageDate argument to start next stage.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePATCH(ServerRequestInterface $request): ResponseInterface
+    {
+        $data          = json_decode((string) $request->getBody(), true);
+        $meta          = $data['Meta'] ?? [];
+        $nextStageDate = $data['NextStageDate'] ?? null;
+
+        $requiredfields = [
+            'CandID',
+            'Visit',
+            'Site',
+            'Battery',
+            'Project',
+        ];
+        $diff           = array_diff($requiredfields, array_keys($meta));
+
+        if (!empty($diff)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Field(s) missing: ' . implode(', ', $diff)
+            );
+        }
+
+        if (!$nextStageDate) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'There is no stage date specified'
+            );
+        }
+
+        $dateSegments = explode("-", $nextStageDate);
+        if (!checkdate((int)$dateSegments[1], (int)$dateSegments[2], (int)$dateSegments[0])) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'The date specified in the POST data is invalid. date must be of the form YYYY-MM-DD'
+            );
+        }
+
+        $sessionID = \NDB_Factory::singleton()->database()->pselectOne(
```

Comment:
To avoid this static call, the `_handlePUT` code is using 
```
        $sessionid = array_search( 
               $meta['Visit'], 
               $this->_candidate->getListOfVisitLabels()
        );
```

But, in this case, the timepoint must exists so `$this->_visit` must not be null. You should still check if `$meta['Visit'] == $this->_visit->getVisitLabel()`

Reply:
Thanks for the pointer. I refactored the validation section.